### ISSUE TITLE
JANGTQ_MPP_NAX=auto appears to produce incorrect prefill logits for MXTQ/hybrid JANGTQ models in vmlx serve

### ISSUE THREAD

**Issue by danielkhalilov** [jjang-ai/vmlx#181]
## Summary

After upgrading from vMLX `1.5.32` to `1.5.49`, an MXTQ / JANGTQ4 model began producing incorrect output through `vmlx serve`, while standalone `model()` forwards on the same token IDs remained correct.

The regression bisects to vMLX `1.5.34`. That version appears to have introduced `_apply_jangtq_mpp_nax_policy()` in `cli.py`, which sets `JANGTQ_MPP_NAX=auto` by default.

With `JANGTQ_MPP_NAX=auto`, prefill logits diverge from standalone. Setting `JANGTQ_MPP_NAX=off` before server startup restores the expected logits and output.

I am filing this as "`JANGTQ_MPP_NAX=auto` appears to trigger incorrect prefill logits" rather than claiming a specific kernel/root cause.

## Environment

| Component | Value |
|---|---|
| Platform | Apple Silicon / macOS / Metal |
| Model | `Qwen3.6-35B-A3B-JANGTQ4-CRACK` |
| Quant/profile | MXTQ / JANGTQ4 |
| Cache shape observed | hybrid: 30 × `ArraysCache` + 10 × `KVCache` layers |
| vMLX last known good | `1.5.32` |
| vMLX first bad | `1.5.34` |
| vMLX latest tested | `1.5.49` |
| `mlx` | `0.31.2` |
| `mlx_lm` | `0.31.3` |
| `jang` | `2.5.29` |

Dependency versions were held constant across the tests. Only the `vmlx` version and/or `JANGTQ_MPP_NAX` value changed.

## Repro outline

The local repro uses:

- Prompt rendered to `1090` tokens
- System prompt + user message: `hello there`
- `enable_thinking=true`
- `temperature=0.0`
- `skip_prefix_cache=true`
- Non-streaming `POST /v1/chat/completions`

Server launch shape:

```bash
vmlx serve <model_dir> --port 8199 --host 127.0.0.1 --log-level WARNING
````

Request shape:

```bash
curl -X POST http://127.0.0.1:8199/v1/chat/completions \
  -H 'Content-Type: application/json' \
  -d '{
    "model": "probe",
    "messages": [
      {"role": "system", "content": "<system prompt rendering to ~1090 total prompt tokens>"},
      {"role": "user", "content": "hello there"}
    ],
    "temperature": 0.0,
    "max_tokens": 5,
    "stream": false,
    "skip_prefix_cache": true
  }'
```

I can provide the exact prompt/token IDs if useful. I am avoiding pasting the full downstream system prompt publicly because it comes from a private downstream application, but the relevant point is that the server and standalone paths were verified to receive byte-identical token IDs.

## Expected behavior

The first token should match standalone `model()` forward and vMLX `1.5.32` server behavior:

| Path                         |        First token |        Logit |
| ---------------------------- | -----------------: | -----------: |
| Standalone `model()` forward | `'The'` / id `760` |    `22.2031` |
| vMLX `1.5.32` server         | `'The'` / id `760` | not captured |

## Actual behavior

With vMLX `1.5.34+` and the default `JANGTQ_MPP_NAX=auto`, the server returns incorrect prefill logits:

| Path                                                  |                                                         First token / output |        Logit | Correct? |
| ----------------------------------------------------- | ---------------------------------------------------------------------------: | -----------: | -------: |
| Standalone `model()` forward, any tested vMLX version |                                                           `'The'` / id `760` |    `22.2031` |      Yes |
| vMLX `1.5.32` server                                  |                                                           `'The'` / id `760` | not captured |      Yes |
| vMLX `1.5.34+` server, `JANGTQ_MPP_NAX=auto`          | raw top-1 `'\n'` / id `198`; generated text can start with `Here` / `Here's` |    `15.9844` |       No |
| vMLX `1.5.34+` server, `JANGTQ_MPP_NAX=off`           |                                                           `'The'` / id `760` |    `22.2031` |      Yes |

## Bisect results

| Version  | Result           |
| -------- | ---------------- |
| `1.5.32` | PASS             |
| `1.5.34` | FAIL — first bad |
| `1.5.36` | FAIL             |
| `1.5.42` | FAIL             |
| `1.5.49` | FAIL             |

No `1.5.33` release appeared to be available on PyPI during the bisect.

## Workaround

Set this before starting `vmlx serve`:

```bash
export JANGTQ_MPP_NAX=off
```

This restores correct output for the tested MXTQ / JANGTQ4 model on vMLX `1.5.34+`.

For downstream safety, Codeinator is currently setting `JANGTQ_MPP_NAX=off` before starting `vmlx serve` for JANGTQ/MXTQ models.

## What was ruled out

* **Prompt rendering / tokenization**: an instrumented server confirmed byte-identical token IDs between the standalone and server paths.
* **Cache reuse**: repro uses `skip_prefix_cache=true`; `cache_detail=None`; fresh cache offsets were observed as zero.
* **Scheduler init**: no model calls occurred during init in the relevant probe, and a post-init standalone forward remained correct.
* **`MLLMModelWrapper`, `mx.new_stream`, and `SingleBatchGenerator` in isolation**: each produced the correct token.
* **Server feature flags**: a 7-config matrix from bare-minimum to full server config failed identically.
* **Downstream wrapper env vars**: a bare-minimum server with no consumer-specific env still failed.

## Scope

This appears specific to MXTQ / hybrid JANGTQ models.

A non-MXTQ JANG model tested downstream, Gemma-4 `JANG_4M`, was not affected by this regression.

**Comment by jjang-ai** [comment]
Fixed in current main and packaged-app guard coverage.

The default MPP policy now disables JANGTQ_MPP_NAX=auto for MXTQ/JANGTQ bundles while still allowing an explicit JANGTQ_MPP_NAX=on for diagnostics. The installed-app probe in build/current-issue181-183-runtime-audit-20260601-qwen3vl-minicpm-mpp.json reports #181 focused_source_slice=pass with installed_app_mpp_auto_policy_disables_mxtq=true.

This is focused to the MPP auto prefill-logit regression described here; broader release is still gated separately by the release manifest.
